pkg/storage/engine/gc.go

```diff
@@ -82,6 +82,9 @@ func (gc GarbageCollector) Filter(keys []MVCCKey, values [][]byte) hlc.Timestamp
 			// higher timestamps immediately preceding this one, since they're above
 			// gc.expiration and are needed for correctness; see #6227.
 			delTS = key.Timestamp
+		} else if i > 0 {
```

Comment:
This doesn't seem correct. If you have the threshold at `t=10` and your keys are at `t=15` and `t=5`, you can't delete the one at `t=5` (for a read at `t=11` is still possible, and it would return the value at `t=5`). If you had another value at, say, `t=4`, then you could delete that, but that's the `i+1 < len(keys)` branch below.

Reply:
You're right. I missed this part. Feel free to reject the PR